sdks/python/apache_beam/runners/sdf_utils.py

```diff
@@ -244,3 +251,63 @@ def get_estimator_state(self):
         return None
 
     return _NoOpWatermarkEstimator()
+
+
+class DeduplictaionWithinDuration(ptransform.PTransform):
```

Comment:
It would be useful to expose a keyed deduplication transform as the common implementation that all use internally so in the future we can turn into a well known URN and then runners could provide optimized deduplication transform implementations.

We want pipeline authors to use this transform and I think it should go into sdks/python/apache_beam/transforms/util.py or into a dedicated file such as sdks/python/apache_beam/transforms/ such as deduplicate.py.
CC: @udim What do you think?

Reply:
I would prefer the second one: `sdks/python/apache_beam/transforms/deduplicate.py`

Reply:
Moved transform to `sdks/python/apache_beam/transforms/deduplicate.py`

Reply:
Unlike Java, we typically don't put transforms in their own module in Python. I would recommend putting it under an existing file such as transforms/util.py.

Reply:
Thanks Udi! Similar to `watermark_estimators.py`, I'm thinking about providing a collection of transforms for deduplicating purpose. Any reason that we don't put a group of similar transforms into one module?

Reply:
No, no reason. Go ahead with using a separate file